rfcs/01780_policy_delegation.md

```diff
@@ -0,0 +1,308 @@
+usually accomplished by reviewing the source code from which the module was
+compiled, by informally or formally ensuring that it matches the user's
+expectations in terms of declassification abilities, and then independently
+compiling it to Wasm in a reproducible way, producing an artefact identical to
+the one that is being verified to which the hash refers.
+
+Admittedly all of this is unlikely to be performed by regular (or even advanced)
+users. Also, it would be unlikely to be useful in practice, because code often
+evolves, and therefore previous data labelled with a specific Wasm module hash
+label would not be able to be declassified by new versions of the code, whatever
+that means in practice (e.g. a version with bug fixes or improved
+functionality).
+
+### Wasm Module Signature
+
+https://github.com/project-oak/oak/blob/573e61e74668db3eb53747f38757b00aa899f1f6/oak_abi/proto/label.proto#L71-L77
+
+In order to facilitate more useful data flows with declassification other than
+just individual Wasm modules identified by a precise hash, Oak also supports
+Wasm module signature labels.
+
+A Wasm module signature principal refers to a public key that is used to sign
+any number of individual Wasm modules (or more precisely, to sign their hashes).
+
+For instance, let's assume there are three Wasm modules, with hashes
+respectively w_0, w_1, w_2. Also there are two (public, secret) key pairs (p_0,
+s_0), (p_1, s_1).
+
+Let's assume that the modules are supposed to implement some declassification
+logic, e.g. counting the number of people in an image given raw pixels. s_0 and
+s_1 are two parties that verify such models (from correctness, privacy and
+security point of view) and sign them if they (respectively) believe that they
+are trustworthy.
+
+The owner of s_0 signs w_0 and w_1 with s_0, while the owner of s_1 signs w_1
+and w_2 with s_1:
+
+```text
+    ┌ w_0
+s_0 │
+    └ w_1 ┐
+          │ s_1
+      w_2 ┘
+```
+
+## Delegation Manifest
+
+The main contribution of this RFC is to specify a mechanism by which parties may
+publish a list of hashes in a standard format and serve it statically over
+HTTPS.
+
+In the example above, the owner of s_0 would need to publish the signed module
+hashes somewhere, and a client would have to know the value of s_0 itself (the
+actual key used for signing the module), which introduces extra complexity and
+parties around the deployment and publishing of new modules.
+
+Here we introduce a delegation mechanism that is based on the publication of a
+manifest file on an HTTPS endpoint.
+
+A Delegation Manifest is identified (and located) by an HTTPS URL pointing to a
+JSON file with the schema described below.
+
+For instance, Google may publish a delegation manifest under
+`https://google.com/oak/manifests/count_people_from_pixels.json` with the
+following contents:
+
+```json
+{
+  "trusted": [
+    {
+      "type": "manifest",
```

Comment:
"manifest" is already a very overloaded concept. I don't have a good suggestion, but it might be good to think of a different name.

Reply:
I agree, let's keep thinking about this.

Reply:
anyways, the emphasis should be on the "delegation", not on the "manifest"